cmd/collector/app/collector.go

```diff
@@ -154,6 +157,10 @@ func (c *Collector) Close() error {
 		c.logger.Error("failed to close span processor.", zap.Error(err))
 	}
 
+	if err := c.tlsCloser.Close(); err != nil {
+		c.logger.Error("failed to close TLS certificate watcher", zap.Error(err))
+	}
+
 	return nil
```

Comment:
not sure why this function keeps logging, would be cleaner to return them via multierr.Wrap, but doesn't need to be in this PR.